include/zephyr/drivers/location.h

```diff
@@ -0,0 +1,315 @@
+	/** The uptime when the position was determined in ticks */
+	k_ticks_t uptime_ticks;
+	/** Accuracy radius in millimeters */
+	uint32_t accuracy;
+};
+
+/**
+ * @brief Bearing data
+ *
+ * The accuracy describes the uncertainty of the bearing in +-
+ * millidegrees with a 68.2% certainty.
+ */
+struct location_bearing {
+	/** Bearing angle relative to True North in millidegrees (0 to 360000) */
+	uint32_t bearing;
+	/** The uptime when the bearing was determined in ticks */
+	k_ticks_t uptime_ticks;
+	/** Accuracy in millidegrees */
+	uint32_t accuracy;
+};
+
+/**
+ * @brief Speed data
+ *
+ * The accuracy describes the uncertainty of the speed in +-
+ * millimeters per second with a 68.2% certainty.
+ */
+struct location_speed {
+	/** Speed over ground in millimeters per second */
+	uint32_t speed;
+	/** The uptime when the speed was determined in ticks */
+	k_ticks_t uptime_ticks;
+	/** Accuracy in millimeters per second */
+	uint32_t accuracy;
+};
+
+/**
+ * @brief Altitude data
+ *
+ * The accuracy describes the uncertainty of the altitude
+ * in +- millimeters with a 68.2% certainty.
+ *
+ * Note that the reference from which the altitude is determined
+ * is specific to the device which produced the data
+ */
+struct location_altitude {
+	/** Altitude in millimeters */
+	int32_t altitude;
+	/** The uptime when the altitude was determined in ticks */
+	k_ticks_t uptime_ticks;
+	/** Accuracy in millimeters */
+	uint32_t accuracy;
+};
+
+/**
+ * @typedef location_position_get_t
+ * @brief API for getting newest position data
+ *
+ * See struct location_position for description of position data
+ */
+typedef int (*location_position_get_t)(const struct device *dev,
+			struct location_position *position);
+
+/**
+ * @typedef location_bearing_get_t
+ * @brief API for getting newest bearing data
+ *
+ * See struct location_bearing for description of bearing data
+ */
+typedef int (*location_bearing_get_t)(const struct device *dev,
+			struct location_bearing *bearing);
+
+/**
+ * @typedef location_speed_get_t
+ * @brief API for getting newest speed data
+ *
+ * See struct location_speed for description of speed data
+ */
+typedef int (*location_speed_get_t)(const struct device *dev,
+			struct location_speed *speed);
+
+/**
+ * @typedef location_altitude_get_t
+ * @brief API for getting newest altitude data
+ *
+ * See struct location_altitude for description of altitude data
+ */
+typedef int (*location_altitude_get_t)(const struct device *dev,
+			struct location_altitude *altitude);
+
+/**
+ * @brief Location events
+ *
+ * Note that multiple events can be set simultaneously
+ */
+#define LOCATION_EVENT_POSITION_UPDATED BIT(0)
+#define LOCATION_EVENT_BEARING_UPDATED BIT(1)
+#define LOCATION_EVENT_SPEED_UPDATED BIT(2)
+#define LOCATION_EVENT_ALTITUDE_UPDATED BIT(3)
+
+/**
+ * @typedef location_event_handler_t
+ * @brief Callback API invoked when a location event is raised
+ *
+ * See struct location_provider_event for description of the event
```

Comment:
Not sure if I missed anything, is `location_provider_event` defined anywhere?

Reply:
That was a remnant from the old design, it has been removed, on top of this, a typedef for the events mask has been added location_events_t along with documentation associating LOCATION_EVENT_ ... with the event handler events argument :)